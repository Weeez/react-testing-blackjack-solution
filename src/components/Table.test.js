import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Table from './Table';
import Hand from './Hand';

configure({adapter: new Adapter()});

describe('<Table />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<Table />);
        expect(wrapper).toMatchSnapshot();
    })

    it('has two hands', () => {
        const wrapper = shallow(<Table />);
        expect(wrapper.find(Hand)).toHaveLength(2);
    })

})