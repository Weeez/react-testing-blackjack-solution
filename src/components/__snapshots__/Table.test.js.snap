// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Hand
          cards={
                    Array [
                              "00",
                              "00",
                            ]
          }
/>,
        <Interface
          computerScore={0}
          deal={[Function]}
          hit={[Function]}
          message="Press 'Deal' to start the game"
          playerScore={0}
          stand={[Function]}
/>,
        <Hand
          cards={
                    Array [
                              "00",
                              "00",
                            ]
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": Array [
            "00",
            "00",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "computerScore": 0,
          "deal": [Function],
          "hit": [Function],
          "message": "Press 'Deal' to start the game",
          "playerScore": 0,
          "stand": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": Array [
            "00",
            "00",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Hand
            cards={
                        Array [
                                    "00",
                                    "00",
                                  ]
            }
/>,
          <Interface
            computerScore={0}
            deal={[Function]}
            hit={[Function]}
            message="Press 'Deal' to start the game"
            playerScore={0}
            stand={[Function]}
/>,
          <Hand
            cards={
                        Array [
                                    "00",
                                    "00",
                                  ]
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": Array [
              "00",
              "00",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "computerScore": 0,
            "deal": [Function],
            "hit": [Function],
            "message": "Press 'Deal' to start the game",
            "playerScore": 0,
            "stand": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": Array [
              "00",
              "00",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
